lib/ansible/plugins/action/synchronize.py

```diff
@@ -216,7 +216,7 @@ def run(self, tmp=None, task_vars=None):
         inventory_hostname = task_vars.get('inventory_hostname')
         dest_host_inventory_vars = task_vars['hostvars'].get(inventory_hostname)
         try:
-            dest_host = dest_host_inventory_vars['ansible_host']
+            dest_host = self._templar.template(dest_host_inventory_vars['ansible_host'])
```

Comment:
this is a partial fix of the symptom, real fix would be to avoid the 'get' above that bypasses hostvars implicit templating, otherwise you need to template every access to dest_host_inventory_vars